### PR TITLE
Course explorer

### LINKED ISSUE #159: Feature Request — Create Course Exploration page
**Job**
Create a new page for Course Exploration

**Describe a possible solution**
A separate page that is designed to help users find courses

**Context**
Adds to #156

### PR DESCRIPTION
**Pull request by IkeLyons**
**Issue**
Closes #159 

**Database Changes/Migrations**
N/A

**Test Modifications**
N/A

**Test Procedure**
Run the site, click on the explore link at the top.

**Photos**
Before: Nothing really to show, essentially no content

After: 
Screen Shot 2020-06-25 at 3 59 16 PM

After in motion (converting from .mov to a .gif reduced quality)
Untitled

**Additional Info**
Sorting the majors based off of the school they are in, and adding all the courses for a major are the next steps.

### CODE REVIEW

File: src/web/src/pages/CourseExplorer.vue [hunk 1]
```diff
@@ -1,23 +1,101 @@
...
+          </b-col>
+        </b-row>
+      </b-col>
+    </b-row>
+  </div>
 </template>
 
 <script>
 import { getCourses } from "../services/YacsService";
+import DepartmentListComponenet from "@/components/DepartmentList";
```
Comment (123joshuawu) [inline src/web/src/pages/CourseExplorer.vue:40]: DepartmentListComponent :)

File: src/web/src/pages/CourseExplorer.vue [hunk 2]
```diff
@@ -1,23 +1,101 @@
 <template>
-  <div>Hi im the course explorer with {{ this.courses.length }} courses</div>
+  <div v-if="ready" class="gridContainer w-100 mb-4">
```
Comment (123joshuawu) [inline src/web/src/pages/CourseExplorer.vue:2]: if you want you can add a `<b-spinner>` if the data is `!ready`, but optional

File: src/web/src/pages/CourseExplorer.vue [hunk 3]
```diff
@@ -1,23 +1,101 @@
...
+        } else {
+          this.deptClassDict[c.department] = [c];
+        }
+      }
+      this.ready = true;
+    });
+  },
+  computed: {
+    coursesChunked() {
+      let arr = Object.keys(this.deptClassDict);
```
Comment (123joshuawu) [inline src/web/src/pages/CourseExplorer.vue:71]: General javascript, prefer `const`

File: src/web/src/components/DepartmentList.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<template>
+  <div class="d-flex flex-column flex-grow-1">
+    <div
+      v-for="(major, index) in majors"
+      :index="index"
```
Comment (123joshuawu) [inline src/web/src/components/DepartmentList.vue:5]: curious what does `:index` do?

File: src/web/src/components/DepartmentList.vue [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+import "@/typedef";
+
+export default {
+  name: "DepartmentList",
+  components: {},
```
Comment (123joshuawu) [inline src/web/src/components/DepartmentList.vue:39]: Can leave out `components`

File: src/web/src/components/DepartmentList.vue [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  props: {
+    majors: Array,
+    id: Number,
+  },
+  computed: {},
```
Comment (123joshuawu) [inline src/web/src/components/DepartmentList.vue:44]: and `computed`

File: src/web/src/components/DepartmentList.vue [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+.major-button {
+  display: inline;
+  background: white;
+  border-style: none;
+  text-align: justify;
+  width: 95%;
+}
+
+.major-button:hover {
+  //important because when you click the color changes and thats annoying
```
Comment (123joshuawu) [inline src/web/src/components/DepartmentList.vue:58]: haha nice one

File: src/web/src/pages/CourseExplorer.vue [hunk 8]
```diff
@@ -1,23 +1,101 @@
...
+.gridContainer {
+  display: inline-grid;
+  grid-template-columns: auto auto;
+  justify-content: center;
+  align-content: center;
+}
+
+.departmentBox {
+  width: 30rem;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
```
Comment (123joshuawu) [inline src/web/src/pages/CourseExplorer.vue:93]: nice it looks sick

File: src/web/src/components/DepartmentList.vue [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      v-for="(major, index) in majors"
+      :index="index"
+      :key="major"
+      role="tablist"
+    >
+      <template>
+        <div class="p-2 mb-1 w-100">
+          <b-button
+            squared
+            v-b-toggle="id + 'accordion-' + index"
```
Comment (123joshuawu) [inline src/web/src/components/DepartmentList.vue:13]: should this be `:v-b-toggle`?

File: src/web/src/components/DepartmentList.vue (hunk 4 above)
Comment (IkeLyons) [reply src/web/src/components/DepartmentList.vue:5]: Huh, I tried removing it to show you the error I was getting when it wasnt there earlier, but now the error is gone. Classic. It was something along the lines of having a unique index attribute for each div helped vue access them. I just removed it as I guess its not needed now

File: src/web/src/components/DepartmentList.vue (hunk 9 above)
Comment (IkeLyons) [reply src/web/src/components/DepartmentList.vue:13]: It actually doesn't work when you add that, I'm not sure why. This was just the way that it was in the vue docs

**Pull request merged**